zwave_js_server/model/controller.py

```diff
@@ -442,10 +449,27 @@ async def async_begin_inclusion(
             require_schema = 11
             # String is assumed to be the QR code string so we can pass as is
             if isinstance(provisioning, str):
+                if (
+                    len(provisioning) < MINIMUM_QR_STRING_LENGTH
+                    or provisioning[:2] != "90"
+                ):
+                    raise ValueError(
+                        f"QR code string must be at least {MINIMUM_QR_STRING_LENGTH} characters "
+                        "long and start with `90`"
+                    )
                 options["provisioning"] = provisioning
             # Otherwise we assume the data is ProvisioningEntry or
             # QRProvisioningInformation
             else:
+                if (
+                    isinstance(provisioning, QRProvisioningInformation)
+                    and provisioning.version == QRCodeVersion.SMART_START
+                ):
+                    await self.async_provision_smart_start_node(provisioning)
```

Comment:
You start something and then raise ?

Reply:
Yes - what this does is stores the provisioning entry in the controller. What would you suggest? Just raise and let the caller do the right thing?

Reply:
Let's just do the right thing.